zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/DataStoreManager.scala

```diff
@@ -161,12 +161,33 @@ class DataStoreManager(metaDataset: String,
     val dropTableName = dropTable.dataset
 
     if(metaData.contains(dropTableName)){
-      metaData.remove(dropTableName)
+      context.child("data-" + dropTableName) match {
+        case Some(datasetActor) => datasetActor ! PoisonPill
```

Comment:
I thought the different way, maybe I am wrong.
The name of the actor, when it is created, depends on the `query.dataset`, which should be the same for all views and the associated dataset. Then in this way, every dataset and all its views will only have one actor.